### PR TITLE
Improve Compiler for diagnostics

### PR DESCRIPTION
**Pull request by KevinEady**
This PR will enhance Bscript with functionality that allows it to be embedded and used for ecompile diagnostics. The end-goal is to use create an "API" that will be used by the Visual Studio extension to implement the LSP protocol without LSP going inside ecompile.

### CODE REVIEW

File: pol-core/bscript/compiler/Compiler.h [hunk 1]
```diff
@@ -2,6 +2,7 @@
 #define POLSERVER_COMPILER_H
 
 #include <memory>
+#include <optional>
```
Comment (turleypol) [inline pol-core/bscript/compiler/Compiler.h:5]: unneeded include?

Comment (KevinEady) [reply pol-core/bscript/compiler/Compiler.h]: Removed!

**Review by frozenblit — APPROVED** [review]

File: pol-core/bscript/compiler/Compiler.cpp [hunk 2]
```diff
@@ -129,12 +138,26 @@ void Compiler::compile_file_steps( const std::string& pathname, Report& report )
   output = generate( std::move( workspace ) );
 }
 
+std::unique_ptr<CompilerWorkspace> Compiler::precompile( const std::string& pathname,
```
[the remark this answers is not in the document]
Comment (KevinEady) [reply pol-core/bscript/compiler/Compiler.cpp]: Addressed!

File: pol-core/bscript/compiler/Report.h [hunk 3]
```diff
@@ -22,7 +65,7 @@ class Report
   {
     fmt::Writer w;
     rec_write( w, std::forward<Args>( args )... );
-    report_error( source_location, w.c_str() );
+    report_error( source_location, w.str() );
   }
 
   // Always put a newline at the end of the message.
```
[the remark this answers is not in the document]
Comment (KevinEady) [reply pol-core/bscript/compiler/Report.h]: Addressed!

**Pull request merged**